pkg/ebpf/tracer.go

```diff
@@ -304,7 +304,7 @@ func (t *Tracer) initPerfPolling() (*bpflib.PerfMap, error) {
 //  • Local DNS (*:53) requests if configured (default: true)
 func (t *Tracer) shouldSkipConnection(conn *ConnectionStats) bool {
 	isDNSConnection := conn.DPort == 53 || conn.SPort == 53
-	if !t.config.CollectLocalDNS && isDNSConnection && conn.Direction == LOCAL {
+	if !t.config.CollectLocalDNS && isDNSConnection {
```

Comment:
Hmm we'll want to make sure we add an additional check to confirm this is a local DNS connection if we're removing `conn.Direction == LOCAL`

Reply:
Possibly looking at the IP to see if it's in the LAN subnets

Reply:
https://github.com/DataDog/datadog-agent/pull/4727/files will fix this.